source/extensions/filters/common/rbac/engine_impl.h

```diff
@@ -25,6 +29,38 @@ using DynamicMetadataKeysSingleton = ConstSingleton<DynamicMetadataKeys>;
 
 enum class EnforcementMode { Enforced, Shadow };
 
+struct ActionContext {
+  bool has_log_;
+};
+
+class Action : public Envoy::Matcher::ActionBase<envoy::config::rbac::v3::Action> {
+public:
+  Action(const std::string& name, const envoy::config::rbac::v3::RBAC::Action action)
+      : name_(name), action_(action) {}
+
+  const std::string& name() const { return name_; }
+  envoy::config::rbac::v3::RBAC::Action action() const { return action_; }
+
+private:
+  const std::string name_;
+  const envoy::config::rbac::v3::RBAC::Action action_;
+};
+
+class ActionFactory : public Envoy::Matcher::ActionFactory<ActionContext> {
+public:
+  Envoy::Matcher::ActionFactoryCb
+  createActionFactoryCb(const Protobuf::Message& config, ActionContext& context,
+                        ProtobufMessage::ValidationVisitor& validation_visitor) override;
+  std::string name() const override { return "action"; }
```

Comment:
nit: name needs to be qualified with `envoy.`

Reply:
Done, changed to `envoy.filters.rbac.action`.